alchemist/alchemist-swingui/src/main/java/it/unibo/alchemist/boundary/gui/effects/DrawNavigationGraph.java

```diff
@@ -106,7 +106,7 @@
                 }
             });
         }
-        if (env instanceof ImageEnvironmentWithGraph) {
+        if (graph == null && env instanceof ImageEnvironmentWithGraph) {
```

Comment:
Why is this on the bottom of the method and not on the beginning? Doesn't it skip one update?

Reply:
It did, fixed.